tests/common/mod.rs

```diff
@@ -0,0 +1,278 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0 OR BSD-3-Clause
+//
+// For some reason clippy & check think that the public methods in this file are not used
```

Comment:
What's the specific complaint? Why don't we see it in firecracker?

Reply:
I really have no idea what's the difference between firecracker and what I am doing here.
Without this I get the following warnings:
```
warning: method is never used: `new`
   --> tests/common/mod.rs:149:5
    |
149 |     pub fn new(first_data: u32) -> Self {
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: `#[warn(dead_code)]` on by default

warning: method is never used: `trigger_all_counters`
   --> tests/common/mod.rs:161:5
    |
161 |     pub fn trigger_all_counters(&mut self) {
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: method is never used: `get_all_counter_values`
   --> tests/common/mod.rs:167:5
    |
167 |     pub fn get_all_counter_values(&self) -> Vec<u64> {
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: method is never used: `new`
   --> tests/common/mod.rs:236:5
    |
236 |     pub fn new() -> Self {
    |     ^^^^^^^^^^^^^^^^^^^^

warning: method is never used: `trigger_event`
   --> tests/common/mod.rs:243:5
    |
243 |     pub fn trigger_event(&self) {
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: method is never used: `clear_event`
   --> tests/common/mod.rs:247:5
    |
247 |     pub fn clear_event(&self) {
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^

warning: method is never used: `counter`
   --> tests/common/mod.rs:251:5
    |
251 |     pub fn counter(&self) -> u64 {
```